llvm/test/Analysis/LoopAccessAnalysis/retry-runtime-checks-after-dependence-analysis-forked-pointers.ll

```diff
@@ -83,10 +83,52 @@ exit:
 define void @dependency_check_and_runtime_checks_needed_select_of_ptr_add_recs(ptr %a, ptr %b, ptr %c, i64 %offset, i64 %n) {
 ; CHECK-LABEL: 'dependency_check_and_runtime_checks_needed_select_of_ptr_add_recs'
 ; CHECK-NEXT:    loop:
-; CHECK-NEXT:      Report: cannot check memory dependencies at runtime
+; CHECK-NEXT:      Memory dependences are safe with run-time checks
```

Comment:
Can you explain where the behavioral change comes from?

Reply:
`%select = select i1 %cmp, ptr %gep.b, ptr %gep.c` will be translated to 2 AddRecs, one for `%gep.b`, and one for `%gep.c`. Before the patch, we would bail out if we need to check dependencies and there are multiple TranslatedPtrs for a source pointer.

With the patch, we now use can determine no-wrap for both of the AddRecs, without needing to look at the IR pointers.